File: my-folder/0121-best-time-to-buy-and-sell-stock/solution.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        minPrice = prices[0]
        maxProfit = 0
        
        for i in range(1, len(prices)):
            if prices[i] < minPrice:
                minPrice = prices[i]
            maxProfit = max(maxProfit, prices[i] - minPrice)
        
        return maxProfit

# prices = [7,1,5,3,6,4]     
#                 i
# minprice = 1
# maxprofit = 5

File: my-folder/0121-best-time-to-buy-and-sell-stock/solution.java
class Solution {
    public int maxProfit(int[] prices) {
        int buy = prices[0];
        int sell = 0;
        int profit = 0;
        for(int i = 1; i < prices.length; i++){
            sell = prices[i];
            buy = Math.min(buy, prices[i]);
            profit = Math.max(profit, sell - buy);
        }

        return profit;
    }
}
